include/ebpf_windows.h

```diff
@@ -49,3 +60,50 @@ typedef enum _ebpf_helper_function
     EBPF_UPDATE_ELEMENT = 2, ///< Update map element.
     EBPF_DELETE_ELEMENT = 3, ///< Delete a map element.
 } ebpf_helper_function_t;
+
+#define EBPF_MAX_GENERAL_HELPER_FUNCTION 0xFFFF
+
+// Supported version numbers for the various eBPF extension data structures.
+#define EBPF_ATTACH_CLIENT_DATA_MAJOR_VERSION_1 1
+#define EBPF_ATTACH_CLIENT_DATA_MINOR_VERSION_0 0
+
+#define EBPF_ATTACH_PROVIDER_DATA_MAJOR_VERSION_1 1
+#define EBPF_ATTACH_PROVIDER_DATA_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_INFORMATION_CLIENT_DATA_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_INFORMATION_CLIENT_DATA_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_TYPE_DESCRIPTOR_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_TYPE_DESCRIPTOR_MINOR_VERSION_0 0
+
+#define EBPF_HELPER_FUNCTION_PROTOTYPE_MAJOR_VERSION_1 1
+#define EBPF_HELPER_FUNCTION_PROTOTYPE_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_INFORMATION_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_INFORMATION_MINOR_VERSION_0 0
+
+#define EBPF_HELPER_FUNCTION_ADDRESSES_MAJOR_VERSION_1 1
+#define EBPF_HELPER_FUNCTION_ADDRESSES_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_SECTION_INFORMATION_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_SECTION_INFORMATION_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_DATA_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_DATA_MINOR_VERSION_0 0
+
+#define EBPF_PROGRAM_SECTION_MAJOR_VERSION_1 1
+#define EBPF_PROGRAM_SECTION_MINOR_VERSION_0 0
+
+/**
+ * @brief Header of an eBPF extension data structure.
+ * Every eBPF extension data structure must start with this header.
+ * New fields can be added to the end of an eBPF extension data structure
+ * without breaking backward compatibility. The minor version field must be incremented in such cases. The major version
+ * field must be updated only if the new data structure is not backward compatible.
+ */
+typedef struct _ebpf_extension_header
+{
+    uint16_t major_version; ///< Major version of the extension data structure.
+    uint16_t minor_version; ///< Minor version of the extension data structure.
```

Comment:
I still see no value in having a minor_version field in this structure.  It's completely redundant with the `size` field in my opinion, and adding it here increases the opportunity for bugs.  In a sense the `size` field IS a minor version already so we don't need a second one.

Reply:
The `size` field will be ambiguous due to compiler padding structs. Either we have to add explicit "reserved" fields to pad or pack at 1 byte and incur perf hit.

Reply:
Not quite.  I'm saying size should NOT be set to sizeof(struct), it should be set to FIELD_END(last_member).  If you do this, there's no reason to have a minor version, and it's less bug prone.

Reply:
If the last field in the struct is a bitfield, and in the next iteration another bitfield is added (such as #3344) ; what macro can be used to detect the change in size?

Reply:
For the case where the bits are always zero until later a use for them is added where the zero value is backwards compatible, then bit fields can be put into a named struct, and then you just use the same FIELD_END macro on the named struct.

If a zero value is not backwards compatible, I will argue that adding such fields into existing bytes is a bug-prone way to do additions over time, so I'd disallow it.  In that case the fact that no macro can be used is a good thing.

Bottom line is that 3344 ought to put bitfields into a named struct that sizeof() can work with.

Reply:
Accepted your proposal. Please check.